src/main/kotlin/org/rust/ide/hints/RsGenericParameterInfoHandler.kt

```diff
@@ -0,0 +1,148 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.hints
+
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.lang.parameterInfo.*
+import com.intellij.openapi.util.TextRange
+import com.intellij.util.containers.nullize
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+
+private const val WHERE_PREFIX = "where "
+
+class RsGenericParameterInfoHandler : ParameterInfoHandler<RsTypeArgumentList, RsGenericPresentation> {
+
+    var curParam = -1
+        private set
+    var hintText = ""
+        private set
+
+    override fun showParameterInfo(element: RsTypeArgumentList, context: CreateParameterInfoContext) {
+        context.highlightedElement = null
+        context.showHint(element, element.textRange.startOffset, this)
+    }
+    
+    // todo: don't disappear in nested generic types
+    override fun updateParameterInfo(parameterOwner: RsTypeArgumentList, context: UpdateParameterInfoContext) {
+        curParam = ParameterInfoUtils.getCurrentParameterIndex(parameterOwner.node, context.offset, RsElementTypes.COMMA)
+        when {
+            context.parameterOwner == null -> context.parameterOwner = parameterOwner
+
+            // occurs e.g. in case of up-down cursor moving
+            context.parameterOwner != parameterOwner -> context.removeHint()
+        }
+    }
+
+    override fun updateUI(p: RsGenericPresentation, context: ParameterInfoUIContext) {
+        hintText = p.presentText
+        val (startOffset, endOffset) =
```

Comment:
Please, be consistent with `if` braces; it is ok not to use them only when `if expression` can be put on one line. In This case you should add braces like 
```kt
            if (hintText.startsWith(WHERE_PREFIX)) {
                0 to WHERE_PREFIX.length 
            } else {
                p.getRange(curParam).startOffset to p.getRange(curParam).endOffset
            }
```